src/test_workflow/benchmark_test/benchmark_create_cluster.py

```diff
@@ -0,0 +1,156 @@
+from test_workflow.integ_test.utils import get_password
+
+
+class BenchmarkCreateCluster(BenchmarkTestCluster):
+    manifest: Union[BundleManifest, BuildManifest]
+    work_dir: str
+    current_workspace: str
+    output_file: str
+    params: str
+    is_endpoint_public: bool
+    cluster_endpoint: str
+
+    """
+    Represents a performance test cluster. This class deploys the opensearch bundle with CDK. Supports both single
+    and multi-node clusters
+    """
+
+    def __init__(
+            self,
+            args: BenchmarkArgs,
+            bundle_manifest: Union[BundleManifest, BuildManifest],
+            config: dict,
+            current_workspace: str
+    ) -> None:
+        super().__init__(args)
+        self.manifest = bundle_manifest
+        self.current_workspace = current_workspace
+        self.output_file = "output.json"
+        role = config["Constants"]["Role"]
+        params_dict = self.setup_cdk_params(config)
+        params_list = []
+        for key, value in params_dict.items():
+            if value:
+                '''
+                TODO: To send json input to typescript code from command line it needs to be enclosed in
+                single-quotes, this is a temp fix to achieve that since the quoted string passed from command line in
+                tesh.sh wrapper script gets un-quoted and we need to handle it here.
+                '''
+                if key == 'additionalConfig':
+                    params_list.append(f" -c {key}=\'{value}\'")
+                else:
+                    params_list.append(f" -c {key}={value}")
+        role_params = (
+            f" --require-approval=never --plugin cdk-assume-role-credential-plugin"
+            f" -c assume-role-credentials:writeIamRoleName={role} -c assume-role-credentials:readIamRoleName={role} "
+        )
+        self.params = "".join(params_list) + role_params
+        self.is_endpoint_public = False
+        self.stack_name = f"opensearch-infra-stack-{self.args.stack_suffix}"
+        if self.manifest:
+            self.stack_name += f"-{self.manifest.build.id}-{self.manifest.build.architecture}"
+
+    def start(self) -> None:
+        command = f"npm install && cdk deploy \"*\" {self.params} --outputs-file {self.output_file}"
+
+        logging.info(f'Executing "{command}" in {os.getcwd()}')
+        subprocess.check_call(command, cwd=os.getcwd(), shell=True)
+        with open(self.output_file, "r") as read_file:
+            load_output = json.load(read_file)
+            self.create_endpoint(load_output)
+        self.wait_for_processing()
+        logging.info("Wait for processing executed")
+
+    def create_endpoint(self, cdk_output: dict) -> None:
+        load_balancer_url = cdk_output[self.stack_name].get('loadbalancerurl', None)
+        if load_balancer_url is None:
+            raise RuntimeError("Unable to fetch the cluster endpoint from cdk output")
+        self.args.cluster_endpoint = load_balancer_url
```

Comment:
Why are we overwriting the args.cluster_endpoint?
Need better way to handle this.

Reply:
Assuming cluster_endpoint is provided in self.args.cluster_endpoint, In cases where manifest and distribution _url are provided this is empty. I will remove the above line. 
`self.cluster_endpoint_with_port` should do the work for us

Reply:
override the self.cluster_endpoint class var here as mentioned in other comments.